src/cli/services/linter/utils/deduplicateResults.ts

```diff
@@ -1,26 +1,28 @@
-import { IRange } from '@stoplight/types/dist';
+import { resultsComparer } from '../../../../formatters/utils/sortResults';
 import { IRuleResult } from '../../../../types';
 
-type Dictionary<T, K extends PropertyKey> = { [key in K]: T };
+export const deduplicateResults = (results: IRuleResult[]): IRuleResult[] => {
+  const filtered: IRuleResult[] = [];
 
-const ARTIFICIAL_ROOT = Symbol('root');
+  const howMany = results.length;
 
-const serializeRange = ({ start, end }: IRange) => `${start.line}:${start.character}:${end.line}:${end.character}`;
-const getIdentifier = (result: IRuleResult) => `${result.path.join('/')}${result.code}${serializeRange(result.range)}`;
+  if (howMany < 2) {
+    return [...results];
+  }
 
-export const deduplicateResults = (results: IRuleResult[]) => {
-  const seen: Dictionary<Dictionary<string, string>, symbol> = {};
+  const sorted = results.sort(resultsComparer);
```

Comment:
I suspect you don't intent to mutate `results`.
```suggestion
  const sorted = [...results].sort(resultsComparer);
```

Reply:
You might need to update some other places in the code to reflect the above change.
I bet certain tests will fail, i.e `src/cli/services/__tests__/linter.test.ts`

Reply:
>I bet certain tests will fail, i.e src/cli/services/__tests__/linter.test.ts

I was expecting them to also fail as well, but the CI server doesn't complain yet. It's possible the current harness doesn't contain enough different broken rules that hit the same line and char range for the "wrongly ordered" results to appear.